### DISCUSSION TITLE
How to use a dynamically reduced denoise strength when processing batch images from animatediff animation

### DISCUSSION THREAD

**Discussion by whmc76** [Comfy-Org/ComfyUI#3807] [Q&A]
How to use a dynamically reduced denoise strength when processing batch images from animatediff animation, so that the animation effect is gradually weakened and transformed into the effect of the original video?
I feel that this should not be the scope of animatediff's responsibility, more should be a modification of the sampler, but I can't find a way to make the denoise of the batch image decrease linearly in a task

**Comment by BrechtCorbeel** [comment]
It's just a loop where you decrease the noise based on the noise strength_total divided by the amount of frames total, so make a node if you know how to.

**Reply by whmc76** [reply]
thanks i will try this

**Reply by Gormint66** [reply]
BrechtCorbeel, thanks for your short answer. I am trying something (maybe exactly) similar to whmc76 but I do not know how to make this with a node. I am struggling getting the frame count of the batch to change the denoise value based on the current frame (or index? I am not sure which would be the right terminology). The problem is when I try Latent Count I get the size of the whole batch not the position (frame) that I am in. How can I get a value of the current frame? Thanks for any help!

**Reply by BrechtCorbeel** [reply]
> BrechtCorbeel, thanks for your short answer. I am trying something (maybe exactly) similar to whmc76 but I do not know how to make this with a node. I am struggling getting the frame count of the batch to change the denoise value based on the current frame (or index? I am not sure which would be the right terminology). The problem is when I try Latent Count I get the size of the whole batch not the position (frame) that I am in. How can I get a value of the current frame? Thanks for any help!

I think you should look at the code of the Ksampler or the VAE, I will look into it myself, but both are involved into making the animation.

VAE is probably too late because the Ksampler does the training, I also do not know if it does that in a whole batch which is then converted, there could be another solution of doing this which is kind of like the following, from a workflow I downloaded:

![image](https://github.com/comfyanonymous/ComfyUI/assets/68238477/433deadf-7906-4c60-b61a-ebc4f0feacea)

You could have the model generate your overall animation at whatever denoise, but then add an additional upscaled ksampler (or not upscaled) and then where you do the difference is that you create a node structure that only does this frame per frame, you then add on this new Ksampler that repeats in a loop until all frames are done with another loop that increase the denoise for each frame or according to an algorythm or pattern you like.

___________________________________

If that second solution is not to your liking it will mainly be up to you in figuring out how to build in this feature into a Ksampler.

**Reply by Gormint66** [reply]
Thank you so much! I will try your suggestions! Much appreciated!

**Reply by whmc76** [reply]
> It's just a loop where you decrease the noise based on the noise strength_total divided by the amount of frames total, so make a node if you know how to.这只是一个循环，您可以根据噪声strength_total除以总帧数来减少噪声，因此如果您知道如何创建节点，请创建一个节点。

This involves some different issues. When I try to locate a batch latent, I find that all the latents become a whole tensor, not a simple sequential sequence as I expected, and because of animatediff, these things are grouped and packaged, probably the context length, but not quite the same.
When I tested this custom ksampler, it worked well to transfer directly using video frames. The frame can be gradually reduced from a high change to the original frame.
But once animatediff is used, the whole result crashes, and there is a huge noise in the picture until the noise disappears, and it seems that only the disordered noise pixels are decreasing.
